pkg/csv2lp/lp_reader.go

```diff
@@ -21,24 +25,35 @@ func LineProtocolFilter(reader io.Reader) *LineProtocolFilterReader {
 	lineReader.LineNumber = 1 // start counting from 1
 	return &LineProtocolFilterReader{
 		lineReader: lineReader,
+		line:       make([]byte, 4096),
+		wrapBuf:    make([]byte, 4096),
 	}
 }
 
 func (state *LineProtocolFilterReader) Read(b []byte) (int, error) {
-	for {
-		bytesRead, err := state.lineReader.Read(b)
-		if err != nil {
-			return bytesRead, err
-		}
-		state.LineNumber = state.lineReader.LastLineNumber
-		buf := b[0:bytesRead]
-		pts, err := models.ParsePoints(buf) // any time precision because we won't actually use this point
-		if err != nil {
-			log.Printf("invalid point on line %d: %v\n", state.LineNumber, err)
-			continue
-		} else if len(pts) == 0 { // no points on this line
-			continue
+	bytesRead, err := state.lineReader.Read(state.line)
+	if err != nil && bytesRead == 0 {
+		return 0, err
+	}
+
+	//read again when we read a partial line at the end of the line reader buffer
+	if bytesRead > 0 && state.line[bytesRead-1] != '\n' {
+		wrapBytesRead, _ := state.lineReader.Read(state.wrapBuf[0:])
+		if wrapBytesRead > 0 {
+			copy(state.line[bytesRead:], state.wrapBuf[0:wrapBytesRead])
+			bytesRead = bytesRead + wrapBytesRead
 		}
-		return bytesRead, nil
 	}
+
+	state.LineNumber = state.lineReader.LastLineNumber
+	pts, err := models.ParsePoints(state.line[0:bytesRead]) // any time precision because we won't actually use this point
+	if err != nil {
+		log.Printf("invalid point on line %d: %v\n", state.LineNumber, err)
+		return 0, nil
+	} else if len(pts) == 0 {
+		return 0, nil
+	}
+
+	copy(b, state.line[0:bytesRead])
```

Comment:
This code has an extra `copy` for every read, and sometimes two extra copies on a line wrap.  

Are those copies necessary? Would it be possible to do all the reads into the passed in buffer, `b`, calling the second read (when there is no `\n` found) `b[bytesRead:]` or something like that?